server.go

```diff
@@ -10,11 +10,22 @@ import (
 	"google.golang.org/grpc"
 )
 
-// PreregisterServices takes a gRPC server and pre-initializes all counters to 0.
-// This allows for easier monitoring in Prometheus (no missing metrics), and should be called *after* all services have
-// been registered with the server.
-func Register(server *grpc.Server) {
-	serviceInfo := server.GetServiceInfo()
+// Server defines the interface that is required for grpc_prometheus to Register.
+type Server interface {
+	GetServiceInfo() map[string]grpc.ServiceInfo
+}
+
+// Register takes a gRPC server and pre-initializes all counters to 0.
+// This allows for easier monitoring in Prometheus (no missing metrics), and
+// should be called *after* all services have been registered with the server.
+func Register(server Server) {
```

Comment:
My thought process here was, ignoring back compat, its somewhat pointless to require the callee to have an implementation that responds to `GetServiceInfo() map[string]grpc.ServiceInfo`; instead, because this library only needs to know the services described by `grpc.ServiceInfo` objects so it can initialize the metrics, it would be more flexible to just allow those to be passed in directly, and the calllees would use whatever facilities they had to get them (`GetServiceInfo()` in the case of `*grpc.Server`, perhaps something else in other implementations). Instead of the current instructions for hooking up, they would have originally been:

```
// Initialize your gRPC server's interceptor.
myServer := grpc.NewServer(
  grpc.StreamInterceptor(grpc_prometheus.StreamServerInterceptor),
  grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor),
)

// Register your gRPC service implementations.
myservice.RegisterMyServiceServer(s.server, &myServiceImpl{})
// After all your registrations, make sure all of the Prometheus metrics are initialized.
grpc_prometheus.Register(myServer.GetServiceInfo())  // <=== this is now requiring only what it cares about
// Register Prometheus metrics handler.    
http.Handle("/metrics", prometheus.Handler())
```

However, in order to not break existing consumers I didn't want to change `Register`, so I figured second best options was to expose the more general register function that most people should use (`RegisterServiceInfo`) and then `Register` is a convenience function for back compat specifically with `*grpc.Server`.